tests/components/humidifier/test_device_action.py

```diff
@@ -0,0 +1,358 @@
+    )
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        "platform": "event",
+                        "event_type": "test_event_turn_off",
+                    },
+                    "action": {
+                        "domain": DOMAIN,
+                        "device_id": "abcdefgh",
+                        "entity_id": "humidifier.entity",
+                        "type": "turn_off",
+                    },
+                },
+                {
+                    "trigger": {
+                        "platform": "event",
+                        "event_type": "test_event_turn_on",
+                    },
+                    "action": {
+                        "domain": DOMAIN,
+                        "device_id": "abcdefgh",
+                        "entity_id": "humidifier.entity",
+                        "type": "turn_on",
+                    },
+                },
+                {
+                    "trigger": {"platform": "event", "event_type": "test_event_toggle"},
+                    "action": {
+                        "domain": DOMAIN,
+                        "device_id": "abcdefgh",
+                        "entity_id": "humidifier.entity",
+                        "type": "toggle",
+                    },
+                },
+                {
+                    "trigger": {
+                        "platform": "event",
+                        "event_type": "test_event_set_humidity",
+                    },
+                    "action": {
+                        "domain": DOMAIN,
+                        "device_id": "abcdefgh",
+                        "entity_id": "humidifier.entity",
+                        "type": "set_humidity",
+                        "humidity": 35,
+                    },
+                },
+                {
+                    "trigger": {
+                        "platform": "event",
+                        "event_type": "test_event_set_mode",
+                    },
+                    "action": {
+                        "domain": DOMAIN,
+                        "device_id": "abcdefgh",
+                        "entity_id": "humidifier.entity",
+                        "type": "set_mode",
+                        "mode": const.MODE_AWAY,
+                    },
+                },
+            ]
+        },
+    )
+
+    set_humidity_calls = async_mock_service(hass, "humidifier", "set_humidity")
+    set_mode_calls = async_mock_service(hass, "humidifier", "set_mode")
+    turn_on_calls = async_mock_service(hass, "humidifier", "turn_on")
+    turn_off_calls = async_mock_service(hass, "humidifier", "turn_off")
+    toggle_calls = async_mock_service(hass, "humidifier", "toggle")
+
+    assert len(set_humidity_calls) == 0
+    assert len(set_mode_calls) == 0
+    assert len(turn_on_calls) == 0
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)